config/initializers/cache_store.rb

```diff
@@ -0,0 +1,44 @@
+require "active_support"
+require "dalli"
+
+module CacheStore
+  CACHE_EXPIRY = 43_200 # 12 hours
+
+  def self.memcached
+    @memcached ||=
+      Dalli::Client.new(
+        ENV["MEMCACHIER_SERVERS"], {
+          username: ENV["MEMCACHIER_USERNAME"],
+          password: ENV["MEMCACHIER_PASSWORD"],
+          expires_in: CACHE_EXPIRY,
+        }
+      )
+  end
+
+  def self.filestore
+    @filestore ||= ActiveSupport::Cache::FileStore.new(".cache")
+  end
+
+  def self.store
+    if ENV["RACK_ENV"] == "production"
```

Comment:
Rather than caching the `@memcached`/`@filestore` variables, I think it makes more sense to cache in `self.store`, i.e.

```ruby
  def self.store
    @store ||= Store.new(cache_instance)
  end

protected

  def self.cache_instance
    if ENV["RACK_ENV"] == "production"
        Dalli::Client.new(ENV["MEMCACHIER_SERVERS"], {
          username: ENV["MEMCACHIER_USERNAME"],
          password: ENV["MEMCACHIER_PASSWORD"],
          expires_in: CACHE_EXPIRY,
        })
    else
      ActiveSupport::Cache::FileStore.new(".cache")
    end
  end
```

It doesn't make much difference in practice, but currently we could make multiple calls to `.store`, which would create a new delegator object every time.

Reply:
I don't think this works, because when the env var changes, we don't need to recreate the @store because it has already been created. It was a bit buggy when running it locally and testing. So I've tweaked a bit but haven't made the changes you've suggested.

Reply:
Fair enough 👍